src/TranslatableTrait.php

```diff
@@ -15,109 +18,104 @@ trait TranslatableTrait
+            return $this->app->_($id, $parameters, $domain, $locale);
+        }
+
+        // case if use TranslatableTrait and translator is defined
+        if ($this->hasTranslator()) {
+            return $this->translator->trans($id, $parameters, $domain, $locale);
+        }
+
+        // FALLBACK : Replicate internal library functionality
+        // if not use TranslatableTrait and is not a Child of App which use TranslatableTrait
+        $translator = new class() implements TranslatorInterface {
+            use TranslatorTrait;
+        };
 
-        return sprintf(...$translated_args);
+        return $translator->trans($id, $parameters, $domain, $locale);
     }
 
     /**
-     * Helper to sprintf a string with domain context.
-     *
-     * _md = multiWithDomain
-     *
-     * @param string        $context    the context domain
-     * @param string        $string     the string to be translated
-     * @param array<array>  $args       an array of array containing args to be translated using methods _ and _d
+     * Return if this object will delegate translations to his own object Translator.
      *
-     * @return string
+     * @return bool
      */
-    public function _md(string $context, string $string, array $args)
+    protected function hasTranslator()
     {
-        $translated_args = [];
-        $translated_args[] = $this->_d($context, $string);
-        $this->_processTranslatedArray($translated_args, $args);
-
-        return sprintf(...$translated_args);
+        return $this->translator instanceof TranslatorInterface;
     }
 
     /**
-     * Process the arguments for sprintf helper.
+     * Return if this object will delegate translations to AppScope.
      *
-     * @param array         $translated_args    reference to translated args in methods _m and _md
-     * @param array<array>  $args               an array of array containing args to be translated using methods _ and _d
+     * @return bool
      */
-    private function _processTranslatedArray(array &$translated_args, array $args)
+    protected function hasTranslatorInAppScope()
     {
-        foreach ($args as $sub_args) {
-
-            if (!is_array($sub_args)) {
-                $sub_args = [$sub_args];
-            }
-
-            switch(count($sub_args))
-            {
-                case 1:
-                    $translated_args[] = $this->_(...$sub_args);
-                    break;
-                case 2:
-                    if(is_numeric($sub_args[1])) {
-                        $translated_args[] = $this->_(...$sub_args);
-                    } else {
-                        $translated_args[] = $this->_d(...$sub_args);
-                    }
-                    break;
-                case 3:
-                    $translated_args[] = $this->_d(...$sub_args);
-                    break;
-            }
-        }
+        return isset($this->app) && $this->app !== $this && method_exists($this->app, '_');
```

Comment:
I think this must be consistent with this:
`            if (!isset($this->app) || !isset($this->app->logger) || !$this->app->logger instanceof \Psr\Log\LoggerInterface) {
`
https://github.com/atk4/core/blob/develop/src/DebugTrait.php#L53